modules/shared/components/organisms/PostCreation/types/IPostCreation.d.ts

```diff
@@ -0,0 +1,37 @@
+import type { EPollType } from './EPollType';
+
+declare namespace IPostCreation {
+  export interface IPostStructure {
+    postType: EPollType.ImagePoll | EPollType.MiniSurvey | EPollType.TextPoll;
+    postCaption: { id: string; body: string };
+    mediaCount: number;
+    groups: {
```

Comment:
create interfaces for group and option and media to make your types more cleaner
```typescript
export interface IPostStructure {
  postType: EPollType;
  postCaption: { id: string; body: string };
  mediaCount: number;
  groups: IGroup[];
  media: IMedia[];
}

export interface IGroup {
  id: string;
  name: string;
  options: IOption[];
  media: IMedia[];
}

export interface IOption {
  id: string;
  body: string;
  media: IMedia[];
}

export interface IMedia {
  id: string;
  file: File;
}
```